corehq/apps/data_interfaces/views.py

```diff
@@ -498,7 +500,8 @@ def post(self, request, *args, **kwargs):
             mode,
             self.domain,
             self.request.couch_user,
-            form_ids_or_query_string
+            form_ids_or_query_string,
+            self.request.can_access_all_locations
         )
```

Comment:
@esoergel i am not sure if something like this already exist and have not added any check as of now.
If need to add one what should be check to ensure all cases are covered. Thinking about:
1. form user_id should be the user's id